source/mainnet/net/references/multi-sig.rst

```diff
@@ -0,0 +1,448 @@
+   in the future and to replace the desktop wallet's file format.
+
+To add additional keys to an existing account on Concordium using the ``concordium-client`` tool, you can either:
+
+.. dropdown:: Add an additional public-private key pair to your current credential
+
+   - Step 1: Create an additional Ed25519 public-private key pair as you feel safe.
+
+   For testing purposes we recommend, the `front-end <https://cyphr.me/ed25519_tool/ed.html>`_ to generate an un-safe
+   additional key pair.
+
+   - Step 2: Find the credential registration ID of your account that we are updating keys for.
+
+   Run the following command where you replace ``<account-name-or-account-address>`` with your account.
+
+   .. code-block:: console
+
+      $concordium-client account show <account-name-or-account-address> --grpc-port <grpc-port-of-server> --grpc-ip <grpc-ip-url-to-node>
+
+   Using the node connection to the hosted testnet node as an example:
+
+   .. code-block:: console
+
+      $concordium-client account show <account-name-or-account-address> --grpc-port 20000 --grpc-ip node.testnet.concordium.com
+
+   This will output something like:
+
+   .. code-block:: console
+
+      Address:                3G5srtaeRvy2uzyaiheY3ZZ6nqTkDg1kRm59xWVnFo8CoEiZ6Y
+      Balance:                1001736.135182 CCD
+      Nonce:                  16
+      Encryption public key:  b14cbfe44a02c6b1f78711176d5f437295367aa4f2a8c2551ee10d25a03adc69d61a332a058971919dad7312e1fc94c5a411f21ab3b0962c6fb664212e41769e83c82288515f16ca72c86f8a03e87531cae46394bffc63ab6168432903d79fb0
+
+      Baking or delegating stake: no
+
+      Credentials:
+      * <credential-registration-id>:
+         - Index: 0
+         - Expiration: Jun 2023
+         - Type: normal
+         - Revealed attributes: none
+
+   We require the ``<credential-registration-id>`` output for the next step.
+
+   - Step 3: Send a transaction to update the keys of your account on-chain.
+
+   .. code-block:: console
+
+      $concordium-client account update-keys --credId <credential-registration-id> --sender <account-name-or-address> ./update-keys.json --grpc-port <grpc-port-of-server> --grpc-ip <grpc-ip-url-to-node>
+
+   Using the node connection to the hosted testnet node as an example:
+
+   .. code-block:: console
+
+      $concordium-client account update-keys --credId <credential-registration-id> --sender <account-name-or-address> ./update-keys.json --grpc-port 20000 --grpc-ip node.testnet.concordium.com
+
+   where ``update-keys.json`` is a file of the following format:
+
+   .. code-block:: json
+      :force:
+
+      {
+         "keys": {
+            "0": {
+               "verifyKey": "<Hex encoding without '0x' of the public key already on-chain>"
+            },
+            "1": {
+               "verifyKey": "<Hex encoding without '0x' of the public key from your newly generated key pair>"
+            }
+         },
+         "threshold": 2
+      }
```

Comment:
Not a comment to the documentation, but would it make sense to have an easier way to do this? Something like

```sh
concordium-client account add-keys \
  --credId <credential-registration-id> \
  --sender <account-name-or-address> \
  --key <verify-key> # --key can have multiple values
```

Just a random thought